### PR TITLE
Zowelog refactor and message IDs

### PR DESCRIPTION
**Pull request by sakeerthy**
Removes unneeded logs, switches printfs to zowelog, adds message IDs

### CODE REVIEW

File: c/alloc.c [hunk 1]
```diff
@@ -139,7 +139,9 @@ char *allocECSA(int size, int key){
   case 2:
     return getmain31Key2(size,241);
   default:
-    printf("PANIC cannot allocate ECSA in key=%d\n",key);
+    if (MALLOC_TRACE_LEVEL >= 1){
+      printf("PANIC cannot allocate ECSA in key=%d\n",key);
```
Comment (1000TurquoisePogs) [inline c/alloc.c:143]: This just may not be needed at all.

File: c/alloc.c [hunk 2]
```diff
@@ -151,7 +153,9 @@ int freeECSA(char *data, int size, int key){
   case 2:
     return freemain31Key2(data,size,241);
   default:
-    printf("PANIC cannot free ECSA in key=%d\n",key);
+    if (MALLOC_TRACE_LEVEL >= 1){
+      printf("PANIC cannot free ECSA in key=%d\n",key);
```
Comment (1000TurquoisePogs) [inline c/alloc.c:157]: This just may not be needed at all.

File: c/alloc.c [hunk 3]
```diff
@@ -410,7 +422,9 @@ char *safeMalloc(int size, char *site){
       abend(MALLOC_ABEND_CODE, MALLOC_ABEND_REASON);
     }
 #endif
-    printf("MALLOC failed, got NULL for size %d at site %s\n",size, site);
+    if (MALLOC_TRACE_LEVEL >= 1){
+      printf("MALLOC failed, got NULL for size %d at site %s\n",size, site);
```
Comment (1000TurquoisePogs) [inline c/alloc.c:426]: This just may not be needed at all, because you will know success by getting a return of NULL.

File: c/alloc.c [hunk 4]
```diff
@@ -453,7 +467,9 @@ char *safeMalloc2(int size, char *site, int *indicator){
     memset(res,0,size);
     *indicator = 0x884;
   } else{
-    printf("MALLOC failed, got NULL for size %d at site %s\n",size, site);
+    if (MALLOC_TRACE_LEVEL >= 1){
+      printf("MALLOC failed, got NULL for size %d at site %s\n",size, site);
```
Comment (1000TurquoisePogs) [inline c/alloc.c:471]: This just may not be needed at all, because you will know success by getting a return of NULL.

File: c/alloc.c [hunk 5]
```diff
@@ -487,7 +502,9 @@ char *safeMalloc31(int size, char *site){
   if (res != NULL){
     memset(res,0,size);
   } else{
-    printf("MALLOC failed, got NULL for size %d at site %s\n",size, site);
+    if (MALLOC_TRACE_LEVEL >= 1){
+      printf("MALLOC failed, got NULL for size %d at site %s\n",size, site);
```
Comment (1000TurquoisePogs) [inline c/alloc.c:506]: This just may not be needed at all, because you will know success by getting a return of NULL.

File: c/alloc.c [hunk 6]
```diff
@@ -525,7 +543,9 @@ static char *safeMalloc64Internal(int size, char *site, long long token){
...
```
Comment (1000TurquoisePogs) [inline c/alloc.c:547]: This just may not be needed at all, because you will know success by getting a return of NULL.

File: c/alloc.c [hunk 7]
```diff
@@ -563,14 +584,15 @@ char *safeMalloc31Key8(int size, char *site){
...
```
Comment (1000TurquoisePogs) [inline c/alloc.c:588]: This just may not be needed at all, because you will know success by getting a return of NULL.

File: c/alloc.c [hunk 8]
```diff
@@ -843,7 +867,9 @@ char *safeMalloc64v2(unsigned long long size, int zeroOut, char *site,
       memset(data, 0, size);
     }
   } else{
-    printf("MALLOC failed, got NULL for size %llu at site %s\n",size, site);
+    if (MALLOC_TRACE_LEVEL >= 1){
+      printf("MALLOC failed, got NULL for size %llu at site %s\n",size, site);
```
Comment (1000TurquoisePogs) [inline c/alloc.c:871]: This just may not be needed at all, because you will know success by getting a return of NULL.

File: c/alloc.c [hunk 9]
```diff
@@ -796,7 +818,9 @@ char *safeMalloc64v2(unsigned long long size, int zeroOut, char *site,
     /* this is to insure that getTCBTokenRC doesn't get compiled out
      * and in case of an error we can access it in the SVC dump */
     if (FALSE) {
-      printf("safeMalloc64v2: job step TCB, rc=%d\n", getTCBTokenRC);
```
Comment (1000TurquoisePogs) [inline c/alloc.c:799]: remove that if FALSE around this

File: c/alloc.c [hunk 10]
```diff
@@ -868,7 +894,9 @@ int safeFree64v2(void *data, unsigned long long size, int *sysRC, int *sysRSN) {
     memset(tcbToken, 0, sizeof(tcbToken));
     int getTCBTokenRC = getJobstepTCBToken(tcbToken);
     if (FALSE) {
-      printf("safeFree64v2: job step TCB, rc=%d\n", getTCBTokenRC);
+      if (MALLOC_TRACE_LEVEL >= 1){
```
Comment (1000TurquoisePogs) [inline c/alloc.c:897]: remove the FALSE. Maybe trace level 2?

File: c/alloc.c [hunk 11]
```diff
@@ -1052,7 +1082,9 @@ char *safeMalloc64v3(unsigned long long size, int zeroOut, char *site,
       memset(data, 0, size);
     }
   } else{
-    printf("MALLOC failed, got NULL for size %llu at site %s\n",size, site);
+    if (MALLOC_TRACE_LEVEL >= 1){
+      printf("MALLOC failed, got NULL for size %llu at site %s\n",size, site);
```
Comment (1000TurquoisePogs) [inline c/alloc.c:1086]: This just may not be needed at all, because you will know success by getting a return of NULL.

File: c/bpxskt.c [hunk 12]
```diff
@@ -582,8 +582,10 @@ Socket *socketAccept(Socket *serverSocket, int *returnCode, int *reasonCode){
          returnCode,
          reasonCodePtr);
   if (returnValue == -1){
-    printf("Failed to accept new socket errno=%d reasonCode=0x%x\n",
-           *returnCode,*reasonCode);
+    if (socketTrace){
+      printf("Failed to accept new socket errno=%d reasonCode=0x%x\n",
```
Comment (1000TurquoisePogs) [inline c/bpxskt.c:586]: This just may not be needed at all, because you will know success by getting a return of NULL.

File: c/bpxskt.c [hunk 13]
```diff
@@ -781,7 +783,9 @@ int tcpIOControl(Socket *socket, int command, int argumentLength, char *argument
           returnCode,
           reasonCodePtr);
   if (returnValue < 0){
-    printf("ioctl failed ret code %d reason 0x%x\n",*returnCode,*reasonCode);
+    if (socketTrace){
+      printf("ioctl failed ret code %d reason 0x%x\n",*returnCode,*reasonCode);
```
Comment (1000TurquoisePogs) [inline c/bpxskt.c:787]: This just may not be needed at all, because you will know success by getting a return of NULL.

File: c/bpxskt.c [hunk 14]
```diff
@@ -936,7 +941,9 @@ int setSocketBlockingMode(Socket *socket, int isNonBlocking,
           reasonCodePtr);
   
   if (returnValue < 0){
-    printf("BPXFCT failed, ret code %d reason 0x%x\n",*returnCode,*reasonCode);
+    if (socketTrace){
+      printf("BPXFCT failed, ret code %d reason 0x%x\n",*returnCode,*reasonCode);
```
Comment (1000TurquoisePogs) [inline c/bpxskt.c:945]: This just may not be needed at all, because you will know success by getting a return of NULL.

File: c/bpxskt.c [hunk 15]
```diff
@@ -1015,7 +1025,9 @@ int socketRead(Socket *socket, char *buffer, int desiredBytes,
           returnCode,
           reasonCodePtr);
   if (returnValue < 0){
-    printf("read failed ret code %d reason 0x%x\n",*returnCode,*reasonCode);
+    if (socketTrace){
+      printf("read failed ret code %d reason 0x%x\n",*returnCode,*reasonCode);
```
Comment (1000TurquoisePogs) [inline c/bpxskt.c:1029]: i think this is not needed because returnvalue, returncode, and reasoncode seem to get back to caller.

File: c/bpxskt.c [hunk 16]
```diff
@@ -1041,7 +1053,9 @@ int socketWrite(Socket *socket, const char *buffer, int desiredBytes,
     int bytesWritten = 0;
     status = rs_ssl_write(socket->sslHandle, buffer, desiredBytes, &bytesWritten);
     if (0 != status) {
-      printf("socketWrite: rs_ssl_write failed with status: %d\n", status);
+      if (socketTrace){
+        printf("socketWrite: rs_ssl_write failed with status: %d\n", status);
```
Comment (1000TurquoisePogs) [inline c/bpxskt.c:1057]: returncode & reasoncode are not set yet, but status is. should we return status to the user so that this print is not needed?

File: c/bpxskt.c [hunk 17]
```diff
@@ -1341,7 +1364,9 @@ int socketSetRemove(SocketSet *set, Socket *socket){
   int sd = socket->sd;
 
   if (sd > set->highestAllowedSD){
-    printf("SD=%d out of range (> %d)\n",sd,set->highestAllowedSD);
+    if (socketTrace){
+      printf("SD=%d out of range (> %d)\n",sd,set->highestAllowedSD);
```
Comment (1000TurquoisePogs) [inline c/bpxskt.c:1368]: not needed because return code already given.

File: c/bpxskt.c [hunk 18]
```diff
@@ -1643,7 +1672,9 @@ int not_main(int argc, char **argv){
     bytesRead = socketRead(socket,readBuffer,200,&errno,&retcode);
     dumpbuffer(readBuffer,bytesRead);
     for (i=0; i<5; i++){
-      printf("SLEEP %d\n",i+1);fflush(stdout);
+      if (socketTrace){
+        printf("SLEEP %d\n",i+1);fflush(stdout);
```
Comment (1000TurquoisePogs) [inline c/bpxskt.c:1676]: may not need a conditional since this appears to be a test function

File: c/discovery.c [hunk 19]
```diff
@@ -282,10 +282,10 @@ static int walkTCBs1(DiscoveryContext *context,
                      void *visitorContext,
                      char *desiredSubsystemName){
   if (FALSE){
-    printf("TCB at depth=%d\n",depth);
-    printf("  tcbntc(sibling): 0x%x\n",tcb->tcbntc);    
-    printf("  tcbotc(parent ): 0x%x\n",tcb->tcbotc);
-    printf("  tcbltc(child  ): 0x%x\n",tcb->tcbltc);
+    zowelog(NULL, LOG_COMP_DISCOVERY, ZOWE_LOG_INFO, "TCB at depth=%d\n",depth);
```
Comment (1000TurquoisePogs) [inline c/discovery.c:285]: if false... how about a debug log instead?

File: c/discovery.c [hunk 20]
```diff
@@ -319,12 +319,12 @@ static void visitSSCTEntry(DiscoveryContext *context,
...
                            char *specificBestName){
   if (context->ssctTraceLevel >= 1){
-    printf("SSCT %.4s at 0x%x specificBestName ptr is 0x%x\n",&(ssctChain->sname),ssctChain,specificBestName);fflush(stdout);
+    zowelog(NULL, LOG_COMP_DISCOVERY, ZOWE_LOG_INFO, "SSCT %.4s at 0x%x specificBestName ptr is 0x%x\n",&(ssctChain->sname),ssctChain,specificBestName);fflush(stdout);
     dumpbuffer((char*)ssctChain,sizeof(SSCT));
   }
   void *usr1 = (void*)ssctChain->ssctsuse;
   if (context->ssctTraceLevel >= 1){
-    printf("user pointer at 0x%x COMMON?=%s\n",usr1,isPointerCommon(gda,usr1) ? "YES" : "NO");
+    zowelog(NULL, LOG_COMP_DISCOVERY, ZOWE_LOG_INFO, "user pointer at 0x%x COMMON?=%s\n",usr1,isPointerCommon(gda,usr1) ? "YES" : "NO");
```
Comment (1000TurquoisePogs) [inline c/discovery.c:327]: fix double-tracing issue

File: c/discovery.c [hunk 21]
```diff
@@ -342,19 +342,19 @@ static void visitSSCTEntry(DiscoveryContext *context,
     if (!memcmp(usrData+4,"ERLY",4)){  
       ERLYFragment *erly = (ERLYFragment*)usr1;
       if (context->ssctTraceLevel >= 1){
-        printf("module name %8.8s\n", erly->erlymodn);
+        zowelog(NULL, LOG_COMP_DISCOVERY, ZOWE_LOG_INFO, "module name %8.8s\n", erly->erlymodn);
```
Comment (1000TurquoisePogs) [inline c/discovery.c:345]: fix double-tracing issue

File: c/discovery.c [hunk 22]
```diff
@@ -687,15 +687,15 @@ int findSessions(DiscoveryContext *context,
         case SESSION_KEY_TYPE_IP4_STRING:
         case SESSION_KEY_TYPE_IP6_STRING:
           {
-            printf("Sesssion remote address is %s\n",addressAsString);
+            zowelog(NULL, LOG_COMP_DISCOVERY, ZOWE_LOG_INFO, "Sesssion remote address is %s\n",addressAsString);
```
Comment (1000TurquoisePogs) [inline c/discovery.c:690]: seems like debug to me?

File: c/http.c [hunk 23]
```diff
@@ -195,7 +195,7 @@ int writeFully(Socket *socket, char *buffer, int len){
   /* this is a sanity check for a bug that once happened on ZOS */
 #ifdef __ZOWE_OS_ZOS
   if (((int)socket) < 0x1000000){
-    printf("*** WARNING *** bad socket pointer!! 0x%x\n",socket);
+	  zowelog(NULL, LOG_COMP_HTTPSERVER, ZOWE_LOG_WARNING, "*** WARNING *** bad socket pointer!! 0x%x\n",socket);
```
Comment (1000TurquoisePogs) [inline c/http.c:198]: may or may not be needed, due to returning 0 saying it all.

File: c/idcams.c [hunk 24]
```diff
@@ -452,7 +452,6 @@ void idcamsPrintCommandOutput(const IDCAMSCommandOutput *idcamsOutput) {
 
   IDCAMSOutputRecord *currentNode = idcamsOutput->firstRecord;
   while (currentNode != NULL) {
-    printf("%.*s\n", currentNode->recordLength, currentNode->record);
```
Comment (1000TurquoisePogs) [inline c/idcams.c:455]: might benefit from debug?

File: c/json.c [hunk 25]
```diff
@@ -47,6 +47,8 @@
 #include "unixfile.h"
 #include "json.h"
 #include "charsets.h"
+#include "logging.h"
```
Comment (1000TurquoisePogs) [inline c/json.c:50]: for this entire file: i'm not sure if zowelog is ok to use here or not. is this a high-enough level library that it is ok?

File: c/le.c [hunk 26]
```diff
@@ -266,7 +266,7 @@ void initRLEEnvironment(RLEAnchor *anchor) {
 
   int recoveryRC = recoveryEstablishRouter(RCVR_ROUTER_FLAG_NONE);
   if (recoveryRC != RC_RCV_OK) {
-    printf("le.c: error - recovery router not established\n");
+    printf("le.c: error - recovery router not established\n"); // TODO: Needs message ID
```
Comment (1000TurquoisePogs) [inline c/le.c:269]: it seems like this rc should be returned, but isnt. is printf really the best we can do here?

File: c/scheduling.c [hunk 27]
```diff
@@ -35,6 +35,8 @@
 #include "recovery.h"
 #include "scheduling.h"
 #include "openprims.h"
+#include "logging.h"
```
Comment (1000TurquoisePogs) [inline c/scheduling.c:38]: i'm like 80% sure that scheduling.c is not something we can use zowelog in. what do others think?

File: c/scheduling.c [hunk 28]
```diff
@@ -207,7 +209,7 @@ static int executeRLETask(RLETask *task) {
   if (recoveryRequired) {
     int establishRC = recoveryEstablishRouter(RCVR_ROUTER_FLAG_NONE);
     if (establishRC != RC_RCV_OK) {
-      printf("error: recovery not established, rc=%d\n", establishRC);
+      zowelog(NULL, LOG_COMP_UTILS, ZOWE_LOG_WARNING, ZCC_LOG_RECOVERY_ERR, establishRC);
```
Comment (1000TurquoisePogs) [inline c/scheduling.c:212]: do we need separate ints to report on taskrc and recoveryrc for the caller to make use of?

File: c/httpfileservice.c [hunk 29]
```diff
@@ -34,6 +34,7 @@
 #include "charsets.h"
 #include "unixfile.h"
 #include "httpfileservice.h"
+#include "zccLogging.h"
```
Comment (ifakhrutdinov) [inline c/httpfileservice.c:37]: Same comment as in `dataservice.c`.

File: c/httpserver.c [hunk 30]
```diff
@@ -62,6 +62,7 @@
 #include "charsets.h"
 #include "impersonation.h"
 #include "httpserver.h"
+#include "zccLogging.h"
```
Comment (ifakhrutdinov) [inline c/httpserver.c:65]: I would avoid using official message IDs in this file. Since this is common code, adding message IDs makes it impossible to reuse in other projects. Perhaps we need to provide hooks, or some message ID parameter when creating the http server in `zss.c`, so it could prepend its messages with the message ID.

Alternatively, all message would be debug. If we start having issues, we enable tracing.

File: c/icsf.c [hunk 31]
```diff
@@ -317,7 +323,9 @@ int icsfDigestFully(char *digestType, char *s, int len){
 	  &md5HashLength,
 	  hash);
 
-  printf("called ICSF MD5 ret=0x%x reason=0x%x\n",returnCode,reasonCode);
+  if (digestTrace){
+    printf("called ICSF MD5 ret=0x%x reason=0x%x\n",returnCode,reasonCode);
+  }
   dumpbuffer(hash,16);
```
Comment (ifakhrutdinov) [inline c/icsf.c:329]: `dumpbuffer` must also be under the `digestTrace`.

File: c/idcams.c (hunk 24 above)
Comment (ifakhrutdinov) [inline c/idcams.c:455]: The only purpose of this function is to print debug information. Why did you remove this printf?

File: c/impersonation.c [hunk 32]
```diff
@@ -258,4 +258,3 @@ int tlsImpersonate(char *userid, char *passwordOrNull, int start, int trace) {
   
   Copyright Contributors to the Zowe Project.
 */
-
```
Comment (ifakhrutdinov) [inline c/impersonation.c:261]: Why changing this?

File: c/jcsi.c [hunk 33]
```diff
@@ -25,6 +25,8 @@
 #include "logging.h"
 #include "csi.h"
 #include "jcsi.h"
+#include "zccLogging.h"
```
Comment (ifakhrutdinov) [inline c/jcsi.c:28]: No need to have official message IDs here. We should make the message debug messages.

File: c/json.c (hunk 25 above)
Comment (ifakhrutdinov) [reply c/json.c:50]: I agree with Sean. We don't need zowelog in this file. We need to look at other json libraries to see how errors are reported.

File: c/le.c [hunk 34]
```diff
@@ -291,4 +291,3 @@ void termRLEEnvironment() {
   
   Copyright Contributors to the Zowe Project.
 */
-
```
Comment (ifakhrutdinov) [inline c/le.c:294]: This is a useless change, please revert.

File: c/le.c (hunk 26 above)
Comment (ifakhrutdinov) [reply c/le.c:269]: I would return some RC and not have an official message here.

File: c/mtlskt.c [hunk 35]
```diff
@@ -23,6 +23,8 @@
 #include "alloc.h"
 #include "utils.h"
 #include "bpxnet.h"
+#include "logging.h"
+#include "zccLogging.h"
```
Comment (ifakhrutdinov) [inline c/mtlskt.c:27]: We should not require official messages in a socket library code. In fact, I think we should avoid any zowelog or messages and return proper rc and rsn codes. Remember, this is library code.

File: c/pdsutil.c [hunk 36]
```diff
@@ -35,6 +35,7 @@
 #include "utils.h"
 #include "logging.h"
 #include "pdsutil.h"
+#include "zccLogging.h"
```
Comment (ifakhrutdinov) [inline c/pdsutil.c:38]: Same comment as in `mtlskt.c`

File: c/recovery.c [hunk 37]
```diff
@@ -2030,4 +2030,3 @@ void recoveryGetABENDCode(SDWA *sdwa, int *completionCode, int *reasonCode) {
   
   Copyright Contributors to the Zowe Project.
 */
-
```
Comment (ifakhrutdinov) [inline c/recovery.c:2033]: Revert this changes.

File: c/scheduling.c (hunk 27 above)
Comment (ifakhrutdinov) [inline c/scheduling.c:38]: Yes, we should avoid doing that. We don't need to change anything since the messages are under the compile time trace flag.

File: c/socketmgmt.c [hunk 38]
```diff
@@ -220,4 +220,3 @@ int sxUpdateTLSInfo(SocketExtension *sext, int onceOnly)
   
   Copyright Contributors to the Zowe Project.
 */
-
```
Comment (ifakhrutdinov) [inline c/socketmgmt.c:223]: Revert this change.

File: c/utils.c [hunk 39]
```diff
@@ -34,6 +34,8 @@
 #include "debug.h"
 #include "printables_for_dump.h"
 #include "timeutls.h"
+#include "logging.h"
```
Comment (ifakhrutdinov) [inline c/utils.c:37]: `utils.c` should be zowelog free. This is utils, so they're reused by everyone, you don't want to make everyone link with logging, le, etc. 

I think we should make sure that the functions that need to print error information can return that error info instead. For example, the SLH code will return NULL if it can't allocate data.

File: c/vsam.c [hunk 40]
```diff
@@ -31,6 +31,7 @@
 #include "utils.h" 
 #include "qsam.h"  
 #include "vsam.h"
+#include "logging.h"
```
Comment (ifakhrutdinov) [inline c/vsam.c:34]: All the messages in this file seem like trace messages. I think we should just keep TRACE. No need to use zowelog.

File: c/xml.c [hunk 41]
```diff
@@ -44,6 +44,8 @@
 #include "zos.h"
 #endif
 #include "xml.h"
+#include "logging.h"
```
Comment (ifakhrutdinov) [inline c/xml.c:47]: This is a low level library code. We should avoid printing any messages.

File: c/zccLogging.c [hunk 42]
```diff
@@ -0,0 +1,56 @@
...
+#else
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <sys/stat.h>
+#include <iconv.h>
+#include <dirent.h>
+#include <errno.h>
+#include <pthread.h>
```
Comment (ifakhrutdinov) [inline c/zccLogging.c:31]: Why so many headers? Can you explain what the purpose of each of them is in this file?

File: c/zosaccounts.c [hunk 43]
```diff
@@ -88,18 +90,16 @@ int gidGetUserInfo(const char *userName,  UserInfo * info,
     if(returnValue == 0) {
 #ifdef METTLE
       /* Do not add a comma */
-      printf("BPXGPN (%s) FAILED: returnValue: %d, returnCode:"
-             " %d, reasonCode: 0x%08x\n",
+      zowelog(NULL, LOG_COMP_ZOS, ZOWE_LOG_SEVERE, ZCC_LOG_METTLE_BPXGPN_ERR,
```
Comment (ifakhrutdinov) [inline c/zosaccounts.c:93]: Same comment as in `dataservice.c`.

File: c/zosfile.c [hunk 44]
```diff
@@ -30,6 +30,8 @@
 #include "charsets.h"
 #include "unixfile.h"
 #include "timeutls.h"
+#include "logging.h"
+#include "zccLogging.h"
```
Comment (ifakhrutdinov) [inline c/zosfile.c:34]: We don't need official message IDs in a library code source file.

File: h/zccLogging.h [hunk 45]
```diff
@@ -0,0 +1,1105 @@
...
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+#ifndef MVD_H_ZCCLOGGING_H_
+#define MVD_H_ZCCLOGGING_H_
```
Comment (ifakhrutdinov) [inline h/zccLogging.h:14]: Why MVD?

File: h/zccLogging.h [hunk 46]
```diff
@@ -0,0 +1,1105 @@
...
+
+#ifndef ZCC_LOG_ID
```
Comment (ifakhrutdinov) [inline h/zccLogging.h:16]: All the message IDs are too specific. We can't afford having hundreds of official message IDs. Each of the will have to be documented with explanation, system programmer response, user response. It will be hard to document what we have here. See https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.3.0/com.ibm.zosmfcore.consoles.messages.help.doc/IZUCN013E.html as example.

File: c/le.c [hunk 47]
```diff
@@ -255,7 +255,7 @@ void deleteRLETask(RLETask *task) {
 #endif
 
 #ifdef __ZOWE_OS_ZOS
-void initRLEEnvironment(RLEAnchor *anchor) {
+int initRLEEnvironment(RLEAnchor *anchor) {
```
Comment (ifakhrutdinov) [inline c/le.c:258]: Changing the return type may potentially break someone's code. Plus returning an RC from recovery.c can be misleading. Given that `recoveryEstablishRouter` shouldn't be failing too often, I think we should leave this alone. In the future we can:
* Abort instead of printing a message since a failure in `recoveryEstablishRouter` isn't something your application would want to continue with
* If needed make another version which returns proper error information

@1000TurquoisePogs what do you think?

**Review by ifakhrutdinov — CHANGES_REQUESTED** [review]

File: c/pdsutil.c [hunk 48]
```diff
@@ -150,7 +149,7 @@ void listDirectory(char *pdsName){
 
   sprintf(filenameBuffer,"//'%s'",pdsName);
   in = fopen(filenameBuffer,"rb");
-  zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "fopen in=0x%x errno=%d\n",in,errno);
+  printf("fopen in=0x%x errno=%d\n",in,errno);
```
[the remark this answers is not in the document]
Comment (sakeerthy) [reply c/pdsutil.c:152]: Your previous comment stated that we should remove zowelogs from pdsutils

[the remark this answers is not in the document]
Comment (sakeerthy) [reply c/pdsutil.c:152]: Change these to debug

[the remark this answers is not in the document]
Comment (ifakhrutdinov) [reply c/pdsutil.c:152]: We need to have some way to return diagnostics info, so it's either rc/rsn or messages. Since we don't have rc/rsn as `mtlskt.c` does, we have to have some messages (we can change the API). Naked `printf`s are bad in library code, so the best we can do is keep `zowelog`. 

In pdsutil.c we should:
* Keep `zowelog` and use `ZOWE_LOG_DEBUG` so they aren't printed without tracing on
* Change `printf` and `dumbuffer` to `zowelog` and `zowedump` with `ZOWE_LOG_DEBUG`

**Review by ifakhrutdinov — APPROVED** [review]
LGTM

**Comment by ifakhrutdinov** [comment]
@1000TurquoisePogs @sakeerthy  can you please squash before merging?

**Comment by 1000TurquoisePogs** [comment]
I think this killed some important logs...
before & after

ZWES1013I ZSS Server has started. Version '1.14.0+20200722'
ZWES1033I Using server config file '~/.zowe/workspace/app-server/serverConfig/server.json'
Host name is DNS or non-numeric, 0
ZWES1015W *** Server has not implemented HTTPS! ***
*** In production, please use 27.0.0.1 or localhost. Server is using '0.0.0.0' ***
ZWES1035I ZSS Server settings: Address='0.0.0.0', port='8542'
ZWES1024I Will not accept JWTs: JWT keystore configuration is missing.
ZWES1000W Privileged server name not provided, falling back to default.
Plugin=org.zowe.configjs, found 0 data service(s)
Plugin=org.zowe.terminal.proxy, found 0 data service(s)
Plugin=org.zowe.terminal.tn3270, found 1 data service(s)
makeDataService begin data service:
added identifier for org.zowe.terminal.tn3270/statediscovery
lookup count=1 name=zosDiscoveryServiceInstaller
Service=org.zowe.terminal.tn3270/statediscovery initializer set internal method=0x3CAFE2D8
makeDataService end
Installing dataservice org.zowe.terminal.tn3270/statediscovery to URI /ZLUX/plugins/org.zowe.terminal.tn3270/services/statediscovery/**
Plugin=org.zowe.zlux.bootstrap, found 0 data service(s)
Plugin=org.zowe.zlux.ng2desktop, found 0 data service(s)
Plugin=org.zowe.zlux.auth.safsso, found 0 data service(s)
Plugin=org.zowe.zlux.sample.angular, found 0 data service(s)
Plugin=org.zowe.zlux.sample.react, found 0 data service(s)
Plugin=org.zowe.zlux.sample.iframe, found 0 data service(s)
Plugin=org.zowe.terminal.vt, found 0 data service(s)
ZWES1039I Installing 'VSAM dataset contents' service...
ZWES1039I Installing 'dataset metadata' service...
ZWES1039I Installing 'dataset contents' service...
ZWES1014I ZIS status - 'Ok' (name='ZWESIS_STD      ', cmsRC='0', description='Ok', clientVersion='2')



after:


ZWES1013I ZSS Server has started. Version '1.15.0+20200901'
ZWES1033I Using server config file '~/.zowe/workspace/app-server/serverConfig/server.json'
ZWES1015W *** Server has not implemented HTTPS! ***
*** In production, please use 27.0.0.1 or localhost. Server is using '0.0.0.0' ***
ZWES1035I ZSS Server settings: Address='0.0.0.0', port='8544'
ZWES1024I Will not accept JWTs: JWT keystore configuration is missing.
ZWES1000W Privileged server name not provided, falling back to default.
ZWES1039I Installing 'VSAM dataset contents' service...
ZWES1039I Installing 'dataset metadata' service...
ZWES1039I Installing 'dataset contents' service...
ZWES1014I ZIS status - 'Ok' (name='ZWESIS_STD      ', cmsRC='0', description='Ok', clientVersion='2')
CEE5206S The signal SIGINT was received.


messages that should still exist:

Plugin=org.zowe.configjs, found 0 data service(s)

Installing dataservice org.zowe.terminal.tn3270/statediscovery to URI /ZLUX/plugins/org.zowe.terminal.tn3270/services/statediscovery/**

because they help us know if the plugins were found & installed correctly.

File: c/dataservice.c [hunk 49]
```diff
@@ -297,7 +297,7 @@ WebPlugin *makeWebPlugin(char *pluginLocation, JsonObject *pluginDefintion, Inte
       }
     }
 
-    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_INFO, "Plugin=%s, found %d data service(s)\n", plugin->identifier, plugin->dataServiceCount);
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG, "Plugin=%s, found %d data service(s)\n", plugin->identifier, plugin->dataServiceCount);
```
Comment (1000TurquoisePogs) [inline c/dataservice.c:300]: This should go back to INFO

File: c/dataservice.c [hunk 50]
```diff
@@ -337,7 +337,7 @@ HttpService *makeHttpDataService(DataService *dataService, HttpServer *server) {
   HttpService *httpService = NULL;
   result = makeHttpDataServiceUrlMask(dataService, urlMask, sizeof(urlMask), server->defaultProductURLPrefix);
   if (result != -1) {
-    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_INFO, "Installing dataservice %s to URI %s\n", dataService->identifier, urlMask);
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG, "Installing dataservice %s to URI %s\n", dataService->identifier, urlMask);
```
Comment (1000TurquoisePogs) [inline c/dataservice.c:340]: this should go back to INFO as well.
